src/MainWindow.cpp

```diff
@@ -0,0 +1,936 @@
+        {
+            // Store the column header
+            QString columnHeader = m_billTableWidget->horizontalHeaderItem(col)->text();
+
+            // If the current column is the bill's funding status and the row is selected
+            if(columnHeader == m_BILL_FUNDING_STATUS_COLUMN_HEADER_STRING && m_billTableWidget->item(row, 0)->isSelected())
+            {
+                // Retrieve the funding status combo box
+                QComboBox *fundedStatusBox;
+                fundedStatusBox = (QComboBox*)m_billTableWidget->cellWidget(row, col);
+
+                // Switch the funding status to funded
+                fundedStatusBox->setCurrentIndex(1);
+            }
+        }
+    }
+}
+
+void MainWindow::defundBillOnClick()
+{
+    // Iterate over the bill table widget rows
+    for(int row = 0; row < m_billTableWidget->rowCount(); row++)
+    {
+        // Iterate over the columns in each row
+        for(int col = 0; col < m_billTableWidget->columnCount(); col++)
+        {
+            // Store the column header
+            QString columnHeader = m_billTableWidget->horizontalHeaderItem(col)->text();
+
+            // If the current column is the bill's funding status and the row is selected
+            if(columnHeader == m_BILL_FUNDING_STATUS_COLUMN_HEADER_STRING && m_billTableWidget->item(row, 0)->isSelected())
+            {
+                // Retrieve the funding status combo box
+                QComboBox *fundedStatusBox;
+                fundedStatusBox = (QComboBox*)m_billTableWidget->cellWidget(row, col);
+
+                // Switch the funding status to not funded
+                fundedStatusBox->setCurrentIndex(0);
+            }
+        }
+    }
+}
+
+void MainWindow::deleteBillOnClick()
+{
+    // Iterate over the bill table widget rows
+    for(int row = 0; row < m_billTableWidget->rowCount(); row++)
+    {
+        // Iterate over the columns in each row
+        for(int col = 0; col < m_billTableWidget->columnCount(); col++)
+        {
+            // Store the column header
+            QString columnHeader = m_billTableWidget->horizontalHeaderItem(col)->text();
+
+            // If the current column is the bill's funding status and the row is selected
+            if(columnHeader == m_BILL_FUNDING_STATUS_COLUMN_HEADER_STRING && m_billTableWidget->item(row, 0)->isSelected())
+            {
+                // Retrieve the funding status combo box
+                QComboBox *fundedStatusBox;
+                fundedStatusBox = (QComboBox*)m_billTableWidget->cellWidget(row, col);
+
+                // If the funding status is marked as funded
+                if(fundedStatusBox->currentText() == m_FUNDED_STRING)
+                {
+                    // Add the bill's amount due back to the amount available by accessing the bill's amount due in the second column
+                    m_totalAmountAvailable+= m_billTableWidget->item(row, col-2)->text().toDouble();
```

Comment:
See if it's worthwhile to add a check here ensuring that there is an amount due in the cell we're attempting to access.

Reply:
There will always be an amount due to start when the bill table widget. If the user attempts to delete a bill after clearing the amount due field, nothing will be subtracted from the total amount available. This is desired behavior so there's no need to add a check at this time. Users should be prevented from inserting a negative value here, that will be added in a future refinement.